### ISSUE TITLE
Default keybindings for loadstate/savestate hide MSX keys

### ISSUE THREAD

**Issue by MBilderbeek** [openMSX/openMSX#1049]
Commit e7e228a8e48ee1373cfe1c056a17efecb492dbfc default keybindings for savestate and loadstate were added:
```
+# keybindings
+bind_default ALT+F8 savestate
+bind_default ALT+F7 loadstate
```
However, all these keys map to MSX keys. ALT maps to GRAPH, F8 maps to STOP and F7 maps to SELECT.

This became a problem when someone (Hi Meits!) tried to bypass the BIOS password which needs the GRAPH+STOP combination and got a savestate instead.. It was impossible to use the key combination without unbinding this default keybinding.

What can we do?
- Remove the keybinding? (Who uses it? Will it lead to more complaints than it solves?)
- Just leave it as it is now as we didn't get complaints in about 8 years?
- Find better binding? Very hard to find unused PC keys... or should we simply use ALT+keys that are not mapped to the MSX keyboard, like F9, F10, F11, F12?

Comments and ideas most welcome.

**Comment by prataczak** [comment]
Hello, SupraKarma1979 from the MSX forum... I think you helped me a bit earlier, and I found this while googling for info.

My two cents - if F12 is not used by MSX, that should be your singular hotkey - it's on the far right, so it's easy to remember/locate, and if you use F12 in combination with other keys, you should be able to map practically anything, since there are a lot of keys. Perhaps F12+alt and F12+shift functionality could be used for special cases where it is ideal to reuse the same key twice (or even three times).

You could make it so that the MSX does not detect any keys being pressed while F12 is being pressed.

Very happy with this emulator, thanks to everyone involved for their contributions.

**Comment by Vampier** [comment]
I have f12 assigned to Pause - it's currently assigned to toggle fullscreen. But I agree that the key is best suited to be re-assigned for your needs.

**Comment by MBilderbeek** [comment]
F12 is currently already assigned by default indeed. But even then, using ALT and SHIFT as modifiers is nice, but these also map to MSX keys (ALT -> GRAPH, SHIFT -> SHIFT).

So the problem is that there are very few free PC keys left. In the end, the user can just bind keys himself and decide whether he wants to accept 'masking' MSX keyboard keys. We can't find something that pleases everyone, I think.

**Comment by prataczak** [comment]
Right on. Well, I haven't had any issues so far (and I probably won't).

Just to be clear though, my idea is to make F12 (or perhaps a different key since that one is already taken) both a hotkey and a disabler (blocking MSX from seeing any other keypresses while it's held). I'd personally being doing stuff like F12+RShift+(whatever other key). Between F12 by itself (+any key) , F12+RShift (+any key), F12+R Alt (+any key), F12+ RCtrl (+any key), you'd never run out of hotkey combinations, and you wouldn't have to worry about any conflicts with the MSX emulation.

Perhaps you'd consider implementing something like that for one of the few remaining keys? I must admit I'm not even sure which keys are even free. :)

I won't complain if you don't though. This is by far the best MSX emulator I've tried.

**Comment by prataczak** [comment]
Hmm... I think I've over-explained it lol. I do that a lot. :)

Basically what I'm asking for is a 'disable MSX keyboard detection while held' hotkey. I guess from there I could figure out a key that isn't being used and map it to that, then set up all of my hotkey combos accordingly.

**Comment by MBilderbeek** [comment]
Yeah, I forgot to respond to that idea. Sounds like a good idea, but I'll discuss it with the other devs. Thanks for sharing it!

**Comment by MBilderbeek** [comment]
@m9710797 what do you think about that idea to get more free keys?
In principle a nice idea, but I wonder if anyone would ever think about it... in other words: it's not intuitive (at all).

**Comment by m9710797** [comment]
I can see the benefits of this idea. But personally I'm not in favor (as in: personally I'll probably continue using the current bindings).

Maybe we better focus on making the key bindings easier to change. I don't remember (and I can't check right now) if we already have a GUI window to help with that.

**Comment by MBilderbeek** [comment]
We don't. And many people miss it.
